### DISCUSSION TITLE
Help Needed: Issue Running DeepForest Model for Tree Species Classification

### DISCUSSION THREAD

**Discussion by Abhijitop25** [weecology/DeepForest#587] [Q&A]
[
 I am trying to install and use DeepForest deep learning model for classifying Indian tree species.
I am reaching out to seek  assistance with an issue I am currently facing during the installation and usage of the model. I have followed the provided instructions, but I encountered an error that I am unable to debug. I believe providing  with more details, including the error message, the multi_csv.csv file, and a sample image (tr.jpg), will help in diagnosing the problem.

Here is a brief description of the issue:

Error Message:

INFO:pytorch_lightning.callbacks.model_summary:
  | Name       | Type                  | Params
-----------------------------------------------------
0 | model      | RetinaNet             | 32.2 M
1 | iou_metric | IntersectionOverUnion | 0     
2 | mAP_metric | MeanAveragePrecision  | 0     
-----------------------------------------------------
31.9 M    Trainable params
222 K     Non-trainable params
32.2 M    Total params
128.675   Total estimated model params size (MB)
/usr/local/lib/python3.10/dist-packages/pytorch_lightning/trainer/connectors/data_connector.py:441: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=1` in the `DataLoader` to improve performance.
/usr/local/lib/python3.10/dist-packages/pytorch_lightning/trainer/connectors/data_connector.py:441: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=1` in the `DataLoader` to improve performance.
Epoch 0: 0%
0/1 [00:00<?, ?it/s]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-29-736c6712a934>](https://localhost:8080/#) in <cell line: 1>()
----> 1 m.trainer.fit(m)
      2 assert m.num_classes == 2

16 frames
[/usr/local/lib/python3.10/dist-packages/torch/_utils.py](https://localhost:8080/#) in reraise(self)
    692             # instantiate since we don't know how to
    693             raise RuntimeError(msg) from None
--> 694         raise exception
    695 
    696 

AttributeError: Caught AttributeError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/worker.py", line 308, in _worker_loop
    data = fetcher.fetch(index)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.10/dist-packages/deepforest/dataset.py", line 106, in __getitem__
    targets["labels"] = image_annotations.label.apply(
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/generic.py", line 5902, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'label'
Files Attached:

multi_csv.csv
tr.jpg
Screenshot of the Error:

multi_csv.csv](https://github.com/weecology/DeepForest/files/13688630/multi_csv.csv)

![tr](https://github.com/weecology/DeepForest/assets/114608202/c58e4c37-a481-4b29-9cb5-3b334c669474)

<img width="568" alt="Screenshot 2023-12-15 195954" src="https://github.com/weecology/DeepForest/assets/114608202/de884bec-e328-43ce-b579-85cbc2d8a5c8">
<img width="569" alt="Screenshot 2023-12-15 200013" src="https://github.com/weecology/DeepForest/assets/114608202/154ac06d-6e65-4089-ac12-fcc5d1f56e2a">
<img width="573" alt="Screenshot 2023-12-15 200034" src="https://github.com/weecology/DeepForest/assets/114608202/125c9c10-ba4f-480a-b82b-1bd25ba9d7d6">
<img width="518" alt="Screenshot 2023-12-15 200056" src="https://github.com/weecology/DeepForest/assets/114608202/b089425c-5cd5-4ac6-a8f0-b2541f40409d">

**Comment by ethanwhite** [comment]
Hi @Abhijitop25 - thanks for the issue. Let's do 2 things to help figure out what the issue is.

1. Can you install the newest development version of the package? If you have git installed on your machine you can do this using:

```sh
pip install git+https://git@github.com/weecology/deepforest.git
```

2. Once you've done that can you run the [multi-species example from the docs](https://deepforest.readthedocs.io/en/latest/multi_species.html) and share the output?

**Comment by ethanwhite** [comment]
@Abhijitop25  - we have now released the fixed version so you can just update to the newest version using `pip install deepforest --upgrade`
